listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -85,3 +96,196 @@ def create_user_recording_recommendation_event(user_name):
     event_data['created'] = event_data['created'].timestamp()
     event_data['event_type'] = event_data['event_type'].value
     return jsonify(event_data)
+
+
+@user_timeline_event_api_bp.route('/user/<user_name>/feed/events', methods=['OPTIONS', 'GET'])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def user_feed(user_name: str):
+    """ Get feed events for a user's timeline.
+
+    :param user_name: The MusicBrainz ID of the user whose timeline is being requested.
+    :type user_name: ``str``
+    :param max_ts: If you specify a ``max_ts`` timestamp, events with timestamps less than or equal to it will be returned
+    :param min_ts: If you specify a ``min_ts`` timestamp, events with timestamps greated than or equal to it will be returned
+    :param count: Optional, number of events to return. Default: :data:`~webserver.views.api.DEFAULT_ITEMS_PER_GET` . Max: :data:`~webserver.views.api.MAX_ITEMS_PER_GET`
+    :statuscode 200: Successful query, you have feed events!
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized, you do not have permission to view this user's feed.
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    """
+
+    user = validate_auth_header()
+    if user_name != user['musicbrainz_id']:
+        raise APIUnauthorized("You don't have permissions to view this user's timeline.")
+
+    db_conn = webserver.create_timescale(current_app)
+    min_ts, max_ts, count, time_range = _validate_get_endpoint_params(db_conn, user_name)
+    if min_ts is None and max_ts is None:
+        max_ts = int(time.time())
+
+    users_following = db_user_relationship.get_following_for_user(user['id'])
```

Comment:
Making n queries, where n is the number of event types, for events from the same DB table is a bad pattern. We should rework this to accomplish all this in one query. This approach is really not future proof.

Reply:
the events live in a number of different tables.

like -> recording_feedback
listen -> listen table in timescale
followers and follow events -> user_relationship
recording_recommendation -> user_timeline_event

I don't think it's realistic to expect to get all events in just the one query.